client/state/reader/follows/reducer.js

```diff
@@ -11,10 +11,77 @@ import {
 	READER_FOLLOW,
 	READER_UNFOLLOW,
 	READER_FOLLOWS_RECEIVE,
+	READER_SUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_SUBSCRIBE_TO_NEW_COMMENT_EMAIL,
+	READER_UPDATE_NEW_POST_EMAIL_SUBSCRIPTION,
+	READER_UNSUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_UNSUBSCRIBE_TO_NEW_COMMENT_EMAIL,
 } from 'state/action-types';
 import { prepareComparableUrl } from './utils';
 import { createReducer } from 'state/utils';
 
+function updatePostSubscription( state, payload, shouldSkipUpdate, newProps ) {
+	const blogId = Number( payload.blogId );
+	const follow = find( state, item => +item.blog_ID === blogId );
+	if ( ! follow ) {
+		return state;
+	}
+
+	if ( shouldSkipUpdate( follow, payload ) ) {
+		return state;
+	}
+
+	return {
+		...state,
+		[ prepareComparableUrl( follow.URL ) ]: {
+			...follow,
+			delivery_methods: {
+				email: {
+					...get( follow, [ 'delivery_methods', 'email' ], {} ),
+					newProps
+				}
+			}
+		}
+	};
+}
+
+function skipNewPostSubscription( current, next ) {
+	return get( current, [ 'delivery_methods', 'email', 'send_posts' ], false ) &&
+			get( current, [ 'delivery_methods', 'email', 'post_delivery_frequency' ] ) === next.deliveryFrequency;
+}
+
+function skipUpdatePostSubscription( current, next ) {
+	return get( current, [ 'delivery_methods', 'email', 'post_delivery_frequency' ] ) === next.deliveryFrequency;
+}
+
+function skipPostUnsubscription( current ) {
+	return get( current, [ 'delivery_methods', 'email', 'send_posts' ], true ) === false;
+}
+
+function updateCommentSubscription( state, blogId, value ) {
+	const follow = find( state, item => item.blog_ID == blogId ); //eslint-disable-line eqeqeq
```

Comment:
eqeq?  after coercing you can change this right?

Also, should you use the selector instead?